payment-manager/src/main/java/com/qpay/paymentmanager/service/impl/WalletServiceImpl.java

```diff
@@ -37,4 +43,41 @@ public void deleteWallet(final long id) {
         walletRepository.deleteById(id);
     }
 
+    @Transactional
+    public WalletEntity paymentWallet(final WalletPayment walletPayment) {
+        isEnoughMoneyInWallet(walletPayment.amount());
+        final var fromWallet = getWalletById(walletPayment.walletIdFrom());
+        final var toWallet = getWalletById(walletPayment.walletIdTo());
+        final var fromWalletBalance = fromWallet.getBalance();
+        final var toWalletBalance = toWallet.getBalance();
+
+        final var currentFromWalletBalance = fromWalletBalance.subtract(walletPayment.amount());
+        isEnoughMoneyInWallet(currentFromWalletBalance);
+        fromWallet.setBalance(currentFromWalletBalance);
+        walletRepository.save(fromWallet);
+
+        final var currentToWalletBalance = toWalletBalance.add(walletPayment.amount());
+        isEnoughMoneyInWallet(currentToWalletBalance);
+        toWallet.setBalance(currentToWalletBalance);
+        walletRepository.save(toWallet);
+
+        return fromWallet;
+    }
+
+    public WalletEntity topUpWallet(final WalletTopUp walletTopUp, final long id) {
+        isEnoughMoneyInWallet(walletTopUp.amount());
+        final var walletEntity = getWalletById(id);
+        final var currentBalanceWallet = walletEntity.getBalance();
+        final var newBalanceWallet = currentBalanceWallet.add(walletTopUp.amount());
```

Comment:
in one method you use `NEWBalanceWallet` in another `CURRENTToWalletBalance` and  `newBALANCEWALLET` in another `currentToWALLETBALANCE`. Sync it